### PR TITLE
fix(gas_price_service): service name and unused trait impl

### PR DESCRIPTION
**Pull request by rymnc**
## Linked Issues/PRs

- none

## Description

Unused trait impl `RunnableService` moved to tests where it's actually used.

## Checklist
- [x] Breaking changes are clearly marked as such in the PR description and changelog
- [x] New behavior is reflected in tests
- [x] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [x] I have reviewed the code myself
- [ ] I have created follow-up issues caused by this PR and linked them here

### After merging, notify other teams

[Add or remove entries as needed]

- [ ] [Rust SDK](https://github.com/FuelLabs/fuels-rs/)
- [ ] [Sway compiler](https://github.com/FuelLabs/sway/)
- [ ] [Platform documentation](https://github.com/FuelLabs/devrel-requests/issues/new?assignees=&labels=new+request&projects=&template=NEW-REQUEST.yml&title=%5BRequest%5D%3A+) (for out-of-organization contributors, the person merging the PR will do this)
- [ ] Someone else?

### CODE REVIEW

File: crates/services/gas_price_service/src/v0/service.rs [hunk 1]
```diff
@@ -194,17 +167,48 @@ mod tests {
...
         Service,
         ServiceRunner,
+        StateWatcher,
     };
     use fuel_core_types::fuel_types::BlockHeight;
     use std::sync::Arc;
     use tokio::sync::mpsc;
 
+    #[async_trait::async_trait]
+    impl<L2, Metadata> RunnableService for GasPriceServiceV0<L2, Metadata>
```
Comment (MitchTurner) [inline crates/services/gas_price_service/src/v0/service.rs:187]: Can we run the task in the tests without the `ServiceRunner`?

Comment (rymnc) [reply crates/services/gas_price_service/src/v0/service.rs:187]: https://github.com/FuelLabs/fuel-core/blob/6979ce6f45a3813ca698f57607db932c8b43f222/crates/services/src/service.rs#L298 needs a `ServiceRunner` to `run_task`, which is private.

Comment (MitchTurner) [reply crates/services/gas_price_service/src/v0/service.rs:187]: Ah. But wait. Can't we just test `UninitializedTask`? That's the true interface that we deal with, not the task anyway.

Comment (rymnc) [reply crates/services/gas_price_service/src/v0/service.rs:187]: yes, but we want to test the behaviour of the task itself here

Comment (rymnc) [reply crates/services/gas_price_service/src/v0/service.rs:187]: there's a pending PR which will have merge conflicts if I add tests for `UninitializedTask` in this PR. let's treat it as a follow up.

Comment (MitchTurner) [reply crates/services/gas_price_service/src/v0/service.rs:187]: That works for me. 

> yes, but we want to test the behaviour of the task itself here

Just if there is no way to run a task without a service, then it's not a public interface and we should treat the service as the interface for that task.

**Review by MitchTurner — APPROVED** [review]

File: crates/services/gas_price_service/src/v0/service.rs (hunk 1 above)
Comment (rymnc) [reply crates/services/gas_price_service/src/v0/service.rs:187]: agreed

**Review by netrome — APPROVED** [review]

**Comment by rymnc** [comment]
waiting on @xgreenx to approve, please don't merge until then

File: crates/services/gas_price_service/src/v0/service.rs (hunk 1 above)
Comment (xgreenx) [inline crates/services/gas_price_service/src/v0/service.rs:187]: I don't see the point of testing code that is not used anywhere

Comment (rymnc) [reply crates/services/gas_price_service/src/v0/service.rs:187]: we're not testing the `RunnableService` it's just a helper for testing the logic within the task.

Comment (xgreenx) [reply crates/services/gas_price_service/src/v0/service.rs:187]: I still don't see why do you need to implement `RunnableService`. The type already implements `RunnableTask`, and you need to test this one=)

Comment (rymnc) [reply crates/services/gas_price_service/src/v0/service.rs:187]: ref https://github.com/FuelLabs/fuel-core/pull/2317#discussion_r1792035909

**Review by xgreenx — APPROVED** [review]

**Review by acerone85 — APPROVED** [review]

**Pull request merged**